rpm/__init__.py

```diff
@@ -64,16 +65,26 @@ def get_sitepackages_and_suffixes(interpreter):
     # try platform-python first (it could be the only interpreter present on the system)
     interpreters = [
         "/usr/libexec/platform-python",
-        f"/usr/bin/python{majorver}",
         f"/usr/bin/python{majorver}.{minorver}",
+        f"/usr/bin/python{majorver}",
     ]
     result = []
     for interpreter in interpreters:
         if not Path(interpreter).is_file():
             continue
+        # skip interpreters with a different major.minor version than the
+        # running interpreter
+        # importing their rpm module has most likely catastrophic consequences
+        i_major, i_minor, _ = re.sub(
+            r"^Python ",
+            "",
+            subprocess.check_output([interpreter, "--version"]).decode().strip(),
+        ).split(".")
+        if (i_major, i_minor) != (majorver, minorver):
+            continue
         sitepackages_and_suffixes = get_sitepackages_and_suffixes(interpreter)
         logger.debug(
-            f"Collected sitepackages and extension suffixes for {interpreter}:\n"
+            f"Collected sitepackages and extension suffixes for {interpreter}:\n"  # noqa: E231
```

Comment:
Why the `noqa` here and below?

Reply:
The problem is that you have an old version of flake8 pinned in the pre-commit config, I have pushed a version bump that should fix this :crossed_fingers: